src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,372 @@
+    {
+        if (!locationInfo.locationInfo.IsNull())
+        {
+            return locationInfo.locationInfo.Value().locationName.data_equal(aLocationName);
+        }
+
+        return false; // name is Null or text sizes do not match
+    }
+
+    /**
+     * This is use for configuring the IsEqual method.
+     * If kIgnoreLocationId is set, the location IDs are ignored when checking for equality.
+     * If kIgnoreMapId is set, the map IDs are ignored when checking for equality.
+     */
+    enum class IsEqualConfig : uint8_t
+    {
+        kIgnoreLocationId = 0x1,
+        kIgnoreMapId      = 0x2,
+    };
+
+    /**
+     * @brief Checks if the given LocationStructureWrapper is equal to this one.
+     * @param aOther The location to compare with.
+     * @param aConfig Set if the location IDs and/or the map IDs should be ignored when checking for equality.
+     * @return True if both locations are equal. False otherwise.
+     */
+    bool IsEqual(const LocationStructureWrapper & aOther, BitMask<IsEqualConfig> aConfig) const
+    {
+        if (!aConfig.Has(IsEqualConfig::kIgnoreLocationId) && (locationID != aOther.locationID))
+        {
+            return false;
+        }
+
+        if (!aConfig.Has(IsEqualConfig::kIgnoreMapId) && (mapID != aOther.mapID))
+        {
+            return false;
+        }
+
+        if (locationInfo.locationInfo.IsNull() != aOther.locationInfo.locationInfo.IsNull())
+        {
+            return false;
+        }
+
+        if (!locationInfo.locationInfo.IsNull())
+        {
+
+            if (!IsNameEqual(aOther.locationInfo.locationInfo.Value().locationName))
+            {
+                return false;
+            }
+
+            if (locationInfo.locationInfo.Value().floorNumber != aOther.locationInfo.locationInfo.Value().floorNumber)
+            {
+                return false;
+            }
+
+            if (locationInfo.locationInfo.Value().areaType != aOther.locationInfo.locationInfo.Value().areaType)
+            {
+                return false;
+            }
+        }
+
+        if (locationInfo.landmarkTag != aOther.locationInfo.landmarkTag)
+        {
+            return false;
+        }
+
+        if (locationInfo.positionTag != aOther.locationInfo.positionTag)
+        {
+            return false;
+        }
+
+        if (locationInfo.surfaceTag != aOther.locationInfo.surfaceTag)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * @brief Get the location name (for logging).
+     * @return The map name as a c style char string.
+     *
+     * @note This is only available through the LocationStructure object (no access from base structure).
+     */
+    const char * name_c_str() const
+    {
+        // note: if name is null, this will return an empty terminated c string - ""
+        return mLocationNameBuffer;
+    }
+
+private:
+    // extra char to allow for terminating 0 ( or name_c_str())
```

Comment:
But we don't terminate.

Reply:
fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/2921b8a8deee83786ce4a6e447e78fb6d614e236).